csharp/src/Ice/SignalTransceiverStream.cs

```diff
@@ -59,14 +81,25 @@ protected ValueTask<T> WaitSignalAsync(CancellationToken cancel = default)
         T IValueTaskSource<T>.GetResult(short token)
         {
             Debug.Assert(token == _source.Version);
-            try
-            {
-                return _source.GetResult(token);
-            }
-            finally
+
+            T result = _source.GetResult(token);
+
+            // Only reset the source if it succeeded. Once the stream got aborted with an exception, we no longer
+            // reset the the source.
+            Debug.Assert(_signaled == 1);
+            _source.Reset();
```

Comment:
The comment says we only reset the source if succeded, but we always call Reset here, is not clear to me how this works

Reply:
Fixed to the following:
```
            // Get the result. This will throw if the stream has been aborted. In this case, we let the
            // exception go through and don't reset the source.
            T result = _source.GetResult(token);

            // Reset the source to allow the stream to be signaled again.
            Debug.Assert(_signaled == 1);
            _source.Reset();
            _signaled = 0;
```